### PR TITLE
Fuhrmanator/issue109

### PR DESCRIPTION
**Pull request by fuhrmanator**
Here's a fix for #109 -- I also updated the usage of Commander and glob.

Lastly, there is a bug fix where functions are being treated as methods (In the component factory). I made it as a separate commit. You may want to treat it separately?

This is a problem especially when a function doesn't belong to a class. The "method" line winds up in a PlantUML file, but it's outside of any class and causes PlantUML to fail. 

image

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -14,7 +14,7 @@ indent_size = 2
 end_of_line = lf
 charset = utf-8
 trim_trailing_whitespace = true
-insert_final_newline = true
+# insert_final_newline = true
```
Comment (bafolts) [inline .editorconfig:17]:
```suggestion
```

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     /* Basic Options */
-    "target": "es5",                         /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
+    "target": "ES2018",                         /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
```
Comment (bafolts) [inline tsconfig.json:4]: Is this change needed for the change to add customization option? Not sure where end users of tplant make use of it today but this could break users depending on es5.

File: src/Factories/ComponentFactory.ts [hunk 3]
```diff
@@ -65,17 +65,19 @@ export function create(fileName: string, node: ts.Node, checker: ts.TypeChecker)
...
-        if (currentNode.name === undefined) {
-          return;
-        }
-        const functionSymbol: ts.Symbol | undefined = checker.getSymbolAtLocation(currentNode.name);
-        if (functionSymbol === undefined) {
-          return;
-        }
-        componentComposites.push(MethodFactory.create(functionSymbol, currentNode, checker));
       }
+      // Functions are not methods in PlantUML! The (fake) methods (that are functions) get placed in the PlantUML outside of a class, and it won't render.
```
Comment (bafolts) [inline src/Factories/ComponentFactory.ts:69]: Not quite following the problem. Is there example input and tsplant output that doesn't render? What doesn't render the output?

Comment (fuhrmanator) [reply src/Factories/ComponentFactory.ts:69]: Take a typescript file with no class that exports a function. The function is modeled as a method. The result in PlantUML will crash.

File: tsconfig.json (hunk 2 above)
Comment (fuhrmanator) [reply tsconfig.json:4]: I had to change it to get the ts of Commander or glob to work

File: src/Factories/ComponentFactory.ts (hunk 3 above)
Comment (fuhrmanator) [reply src/Factories/ComponentFactory.ts:69]: I'm trying to find a simple example to reproduce the problem. I put an image in the PR that shows how it makes PlantUML crash. In that case, `indexOfSplitArray` is defined here: https://github.com/fuhrmanator/FamixTypeScriptImporter/blob/62191c407829e2a00f72886df3bf4c85298299cf/src/famix_functions/famix_functions_index.ts#L99

That function is defined in a module (not in a class). First, functions are not methods in UML (in principle, a TypeScript function can't be represented in a UML class diagram). Second, this function is put in a PlantUML outside of a class (because it's not defined in a class). This makes PlantUML crash with a syntax error. Perhaps I'm misunderstanding how you wanted that to work?

I will try to find a simple way to reproduce the problem.

Comment (fuhrmanator) [reply src/Factories/ComponentFactory.ts:69]: Here's a simple example:

**mySample.ts**
```typescript
export class Foo {
  a(): void {}
}
// a function is not a method in UML
export function f () {
  return "Hello, world!";
}
```

```bash
tplant -i mySample.ts
@startuml
class Foo {
    +a(): void
}
+f(): string
@enduml
```

The `+f(): string` should not be in the PlantUML (a function is not a method). Furthermore, since it's outside a class, it crashes PlantUML with a syntax error.

This bug didn't exist in 3.1.0 (as far as I can tell), and I believe it's caused by the code in the factory that I commented out. Hope this is clear now.

Comment (fuhrmanator) [reply src/Factories/ComponentFactory.ts:69]: FYI I use tplant in a GitHub action to draw the diagram of the API at https://github.com/fuhrmanator/FamixTypeScriptImporter?tab=readme-ov-file#typescript-metamodel-api-documentation-visualization 

I set the Action to use `tplant` 3.1.0 and the problem with the syntax error in PlantUML doesn't happen anymore.

**Review by bafolts — APPROVED** [review]
